pkg/resource/endpoint/manager_test_suite_test.go

```diff
@@ -0,0 +1,131 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package endpoint
+
+import (
+	"errors"
+	"fmt"
+	
+	mocksvcsdkapi "github.com/aws-controllers-k8s/sagemaker-controller/mocks/aws-sdk-go/sagemaker"
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	"github.com/ghodss/yaml"
+	svccommon "github.com/aws-controllers-k8s/sagemaker-controller/pkg/common"
+	"github.com/aws-controllers-k8s/sagemaker-controller/pkg/testutil"
+	acktypes "github.com/aws-controllers-k8s/runtime/pkg/types"
+	svcsdk "github.com/aws/aws-sdk-go/service/sagemaker"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"path/filepath"
+	"testing"
+	ctrlrtzap "sigs.k8s.io/controller-runtime/pkg/log/zap"
+	ackmetrics "github.com/aws-controllers-k8s/runtime/pkg/metrics"
+	"go.uber.org/zap/zapcore"
+)
+
+// provideResourceManagerWithMockSDKAPI accepts MockSageMakerAPI and returns pointer to resourceManager
+// the returned resourceManager is configured to use mockapi api.
+func provideResourceManagerWithMockSDKAPI(mockSageMakerAPI *mocksvcsdkapi.SageMakerAPI) *resourceManager {
+	zapOptions := ctrlrtzap.Options{
+		Development: true,
+		Level:       zapcore.InfoLevel,
+	}
+	fakeLogger := ctrlrtzap.New(ctrlrtzap.UseFlagOptions(&zapOptions))
+	return &resourceManager{
+		rr:           nil,
+		awsAccountID: "",
+		awsRegion:    "",
+		sess:         nil,
+		sdkapi:       mockSageMakerAPI,
+		log:          fakeLogger,
+		metrics:      ackmetrics.NewMetrics("sagemaker"),
+	}
+}
+
+// TestEndpointTestSuite runs the test suite for endpoint
+func TestEndpointTestSuite(t *testing.T) {
```

Comment:
Is there a way to share this file across resources if we provide the whole path to the test_suite.yaml file ?